drivers/can/can_mcp251xfd.c

```diff
@@ -1387,7 +1477,8 @@ static inline int mcp251xfd_init_rx_fifo(const struct device *dev)
 	uint32_t *reg = mcp251xfd_get_spi_buf_ptr(dev);
 	uint32_t tmp;
 
-	tmp = MCP251XFD_REG_FIFOCON_TFNRFNIE | MCP251XFD_REG_FIFOCON_FRESET;
+	tmp = MCP251XFD_REG_FIFOCON_TFNRFNIE | MCP251XFD_REG_FIFOCON_RXOVIE |
```

Comment:
Shouldn't `MCP251XFD_REG_FIFOCON_RXOVIE` be guarded by `CONFIG_CAN_STATS`?

Reply:
The RXOFIE at the fifo doesn't matter if CINT.RXOVIE is guarded.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/17433307/f3313513-e21e-454c-b50a-af63ff488c23)

My initial thoughts were not to add it to avoid cluttering the code with unnecessary ifdefs. I don't mind adding it back if you think it will avoid confusion in the future.

Reply:
I think guarding it or adding a comment would avoid future confusion.